core/src/main/java/com/graphhopper/routing/weighting/custom/CustomProfile.java

```diff
@@ -34,12 +34,13 @@ public CustomProfile(Profile profile) {
     public CustomProfile(String name) {
         super(name);
         setWeighting(CustomWeighting.NAME);
+        setCustomModel(new CustomModel());
     }
 
     public CustomProfile setCustomModel(CustomModel customModel) {
-        customModel.internal();
+        if (customModel != null)
+            customModel.internal();
         getHints().putObject(CustomModel.KEY, customModel);
-        getHints().putObject("custom_model_files", Collections.emptyList());
```

Comment:
I think there is no need for this as CustomProfile is basically independent from the `custom_model_files` hint. It is only used in `config.yml` to specify the custom model indirectly using files (a single method in GraphHopperManaged).